### PR TITLE
chore: add gosec

### PR DESCRIPTION
**Pull request by RafilxTenfen**
- Fix some errors checks
- Add gosec to CI
- Add gosec to makefile 
  - `gosec` runs in docker
  - `gosec-local` runs for local gosec assuming it is available in $PATH

Closes: #666

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -340,6 +341,18 @@ format:
...
+###############################################################################
+###                                Gosec                                    ###
+###############################################################################
+
+gosec:
+	$(DOCKER) run --rm -it -w /$(PROJECT_NAME)/ -v $(CURDIR):/$(PROJECT_NAME) securego/gosec -exclude=G404 -exclude-generated -exclude-dir=/$(PROJECT_NAME)/testutil -exclude-dir=/$(PROJECT_NAME)/test -conf /$(PROJECT_NAME)/gosec.json /$(PROJECT_NAME)/...
+
+gosec-local:
+	gosec -exclude=G404 -exclude-generated -exclude-dir=$(CURDIR)/testutil -exclude-dir=$(CURDIR)/test  -conf $(CURDIR)/gosec.json $(CURDIR)/...
+
```
Comment (SebastianElvis) [inline Makefile:353]: Why do we exclude G404 sec rule?

File: x/btclightclient/keeper/keeper.go [hunk 2]
```diff
@@ -47,10 +48,21 @@ func NewKeeper(
 	}
 }
 
-func (k Keeper) Logger(ctx sdk.Context) log.Logger {
+func Logger(ctx sdk.Context) log.Logger {
 	return ctx.Logger().With("module", fmt.Sprintf("x/%s", types.ModuleName))
 }
 
+// EmitTypedEventWithLog emits an event and logs if it errors.
+func EmitTypedEventWithLog(ctx context.Context, evt proto.Message) {
```
Comment (SebastianElvis) [inline x/btclightclient/keeper/keeper.go:56]: Can we 

- Tie these two functions under `Keeper`?
- Make `EmitTypedEventWithLog` private as it's not supposed to be called by external callers?

**Review by SebastianElvis — APPROVED** [review]
That's a great move!

File: Makefile (hunk 1 above)
Comment (RafilxTenfen) [reply Makefile:353]: G404 does not let us use `math/rand` as import, but some functions on the cosmos-sdk itself ask for that 
In our case, mostly in the `app/test_helpers.go` 

I could also exclude this file from the gosec search if you prefer than exclude the G404 rule

File: x/btclightclient/keeper/keeper.go (hunk 2 above)
Comment (RafilxTenfen) [reply x/btclightclient/keeper/keeper.go:56]: I agree to make it private, but I do not see a reason to tie to the keeper if they don't use any information from the keeper itself

Comment (RafilxTenfen) [reply x/btclightclient/keeper/keeper.go:56]: I can move these two functions to `x/btclightclient/keeper/utils.go`

Comment (RafilxTenfen) [reply x/btclightclient/keeper/keeper.go]: https://github.com/babylonchain/babylon/pull/667/commits/b2e027e0bb2b1e7c631c63bd7f939ba9c65bbc81

File: Makefile (hunk 1 above)
Comment (SebastianElvis) [reply Makefile:353]: If those are test helpers then probably excluding the files is better? G404 is about checking the usage of vulnerable crypto primitives so imo is a meaningful check

File: x/btclightclient/keeper/keeper.go (hunk 2 above)
Comment (SebastianElvis) [reply x/btclightclient/keeper/keeper.go]: > I agree to make it private, but I do not see a reason to tie to the keeper if they don't use any information from the keeper itself

Not a strong requirement, but rather a convention -- you could see other keepers have `Logger` function. No strong objection though

File: Makefile (hunk 1 above)
Comment (RafilxTenfen) [reply Makefile:353]: agreed

File: x/btclightclient/keeper/keeper.go (hunk 2 above)
Comment (RafilxTenfen) [reply x/btclightclient/keeper/keeper.go]: made sense, done at https://github.com/babylonchain/babylon/pull/667/commits/41a955a3ba19d5411f0628619181f034e7ea32b9

File: Makefile (hunk 1 above)
Comment (RafilxTenfen) [reply Makefile]: turns out we were not even using that func '-' 
Removed at https://github.com/babylonchain/babylon/pull/667/commits/368df2e40e2fd5bc9f591db9ed708ef7498d31b8

**Pull request merged**